### PR TITLE
Dashboard action menu

### PR DESCRIPTION
**Pull request by ptbrowne**
Add action menu for layout mode allowing the user to go back to the chart
and go back to the configure mode.
Align the menu used in the configure mode to use the ArrowMenu to have
coherence between the views.

### CODE REVIEW

File: app/configurator/components/add-new-dataset-panel.tsx [hunk 1]
```diff
@@ -78,20 +78,28 @@ export const AddNewDatasetPanel = () => {
   const [dataSetIri, setDataSetIri] = useState("");
   const addNewDataset = useAddChartConfigBasedOnNewDataset();
 
+  const handleClose = useEventCallback(() => {
+    close();
+    setTimeout(() => {
+      // Give time to the drawer to close before resetting the dataset IRI
+      setDataSetIri("");
+    }, 500);
+  });
```
Comment (bprusinowski) [inline app/configurator/components/add-new-dataset-panel.tsx:87]: Nit: Wondering if this shouldn't be in sync with (I assume) framer's transition duration on drawer close?

**Review by bprusinowski — APPROVED** [review]
LGTM! Wondering if the action menu shouldn't be there in Tab layout too? 👀

File: app/configurator/configurator-state.tsx [hunk 2]
```diff
@@ -1513,7 +1519,7 @@ const reducer_: Reducer<ConfiguratorState, ConfiguratorStateAction> = (
       }
       break;
     case "CHART_CONFIG_REMOVE":
-      if (isConfiguring(draft)) {
+      if (isConfiguring(draft) || isLayouting(draft) || isPublishing(draft)) {
```
[the remark this answers is not in the document]
Comment (ptbrowne) [reply app/configurator/configurator-state.tsx:1522]: Not yet, fixed.

File: app/configurator/configurator-state.tsx [hunk 3]
```diff
@@ -1581,7 +1592,13 @@ const reducer_: Reducer<ConfiguratorState, ConfiguratorStateAction> = (
           Object
         );
       }
+      return draft;
 
+    case "CONFIGURE_CHART":
+      const newDraft = transitionStepPrevious(draft, "CONFIGURING_CHART");
+      Object.assign(draft, newDraft);
+      assert(isConfiguring(draft), "Should be configuring after edit");
```
[the remark this answers is not in the document]
Comment (ptbrowne) [reply app/configurator/configurator-state.tsx:1600]: Now, changing the chart key directly on the newDraft, and returning the newDraft, simpler 😅

File: app/configurator/components/add-new-dataset-panel.tsx (hunk 1 above)
Comment (ptbrowne) [reply app/configurator/components/add-new-dataset-panel.tsx]: Thanks for your vigilance, I used the SlideProps.onExited callback for this now :)

**Pull request merged**